File: Assets/Scripts/CSharp2Lua/LuaCallEnum.cs
using System;
using UnityEngine;

public enum TestEnum
{
    LOL = 0,
    Dota2 = 1,
    Overwatch = 2,
}

public class LuaCallEnum : MonoBehaviour
{
    private void Start()
    {
        XLuaEnv.Instance.DoString("CSharp2Lua/LuaCallEnum");
    }

    private void OnDestroy()
    {
        XLuaEnv.Instance.Dispose();
    }
}

File: DataPath/Lua/CSharp2Lua/LuaCallEnum.lua
-- Lua调用枚举

-- CS.命名空间.枚举名.枚举值
-- 枚举获取的是userdata自定义数据类型
print(CS.TestEnum.LOL);
print(CS.TestEnum.Dota2);

print(type(CS.TestEnum.Overwatch));

-- 转换获取枚举值
print(CS.TestEnum.__CastFrom(0));
print(CS.TestEnum.__CastFrom("Dota2"));